src/cadenza.js

```diff
@@ -608,6 +616,25 @@ function validGeometryType(/** @type string */ value) {
   ].includes(value);
 }
 
+/**
+ * @typedef {string} TableParts - An enumeration of the parts that compose an exported data table
+ * (columns, values, totals).
+ */
+
+const TableParts = /** @type {Record<string, TableParts>} */ {
+  COLUMNS: 'columns',
+  VALUES: 'values',
+  TOTALS: 'totals',
+};
+
+function assertSupportedTableParts(/** @type TableParts[] */ parts) {
+  assert(Array.isArray(parts), `'parts' parameter must be an array`);
+  assert(
+    parts.every((p) => Object.values(TableParts).includes(p)),
+    `Invalid 'parts' parameter value: ${parts}`,
+  );
+}
```

Comment:
Proposal: Make the values part of the type definition instead of only mentioning them in the docs.
```typescript
/** @typedef {'columns' | 'values' | 'totals'} TablePart - A part of a table to export */

function assertValidTablePart(/** @type TablePart */ value) {
  assert(
    [ 'columns', 'values', 'totals' ].includes(value),
    `Invalid table part: ${value}`);
}

...

parts.forEach(assertValidTablePart);
```